src/helpers/provider.ts

```diff
@@ -0,0 +1,6 @@
+import { JsonRpcProvider } from '@ethersproject/providers';
+
+export const isSmartContractWallet = async (user: string, provider: JsonRpcProvider) => {
+  const code = await provider.getCode(user);
```

Comment:
unfortunatellyafter EIP7702 this is not enough. You can have EOA with code now :) 

Please, see https://github.com/cowprotocol/cowswap/pull/5735